docs/DeveloperGuide.md

```diff
@@ -1,34 +1,561 @@
+
+The checkstyle configurations is in `<ROOT>/config/checkstyle/` by default. Here is some information on [how to use the
+Checkstyle plugin with IntelliJ IDEA](https://se-education.org/guides/tutorials/checkstyle.html).
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+The target user profile for ra.VI is described by the following:
+* A student of NUS (a freshman in particular)
+* Has a need to manage their school related tasks, classes and notes
+* Prefers desktop apps over other types
+* Types fast
+* Prefers typing to mouse interactions
+* Reasonably comfortable using CLI apps
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+A common problem amongst freshmen is the inability to organise all the incoming information.  
+NUS places a focus on taking responsibility for your own learning, so it might be a tough transition from tertiary education.  
+A lot of students miss lessons, assignments, and even exams, just because they're struggling to adapt to the new
+environment.
+
+ra.VI helps students to manage their school-related information in a compact, stripped-down interface that does not bombard them with too much information.  
+When you receive your modules and lessons, simply enter them into ra.VI as they arrive. ra.VI will keep track of all of it
+for you.  
+You can create tasks, give them deadlines, and tag them to certain modules. You can see all of your tasks and deadlines at a glance.  
+You can even write and save your notes in ra.VI, uncluttering your work environment even further.
+
+ra.VI is even integrated with NUSMods, bringing its comprehensive library of information to your fingertips.  
+All the above features are wrapped in a compact, no-frills command-line interface. No confusing menus and dropdowns
+to distract you; only simple commands to give you what you want.
+
+## Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+## Top level classes
+
+This is a class diagram of the top-level of ra.Vi.  
+The classes depicted here are those which are direct dependencies of the main class Duke.  
+The various dependencies of the classes depicted here are not shown to avoid cluttering, and are described in later sections.  
+
+Duke calls the main class (carried over from the legacy codebase). The main class holds the main loop.  
+Most classes used by the main class are static in nature and do not need to be instantiated. 
+
+The Command and CommandResult objects are dependencies of Executor in addition to Duke. Executor can be
+viewed as a simple layer of abstraction on top of Command and CommandResult to facilitate the execution of user
+commands. Command is a dependency of Parser as Parser creates Command objects to return to the main loop.
+
+![UML class diagram for Main Class](https://github.com/AY2021S1-CS2113T-T09-2/tp/blob/master/docs/diagrams/MainClassDiagram.png?raw=true)
```

Comment:
Could add a picture caption for the Main Class object at the bottom of the picture.